File: Polynom/test/TestPolynom.java
import exceptions.WrongInputSizeException;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestPolynom {

    @Test
    void testInput0() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {3, 0, 0, 0, 0});
        assertEquals("f(x) = 3", polynom.toString());
    }

    @Test
    void testInput1() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {3, 10, 0, 0, 0});
        assertEquals("f(x) = 10x + 3", polynom.toString());
    }

    @Test
    void testInput2() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {4, 5, 2, 0, 0});
        assertEquals("f(x) = 2x^2 + 5x + 4", polynom.toString());
    }

    @Test
    void testInput3() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {4, 5, 2, 7, 0});
        assertEquals("f(x) = 7x^3 + 2x^2 + 5x + 4", polynom.toString());
    }

    @Test
    void testInput4() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {4, 5, 2, 8, 9});
        assertEquals("f(x) = 9x^4 + 8x^3 + 2x^2 + 5x + 4", polynom.toString());
    }

    @Test
    void testMinus() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {4, 5, -2, 8, 9});
        assertEquals("f(x) = 9x^4 + 8x^3 - 2x^2 + 5x + 4", polynom.toString());
    }

    @Test
    void testMinusAtTheStart() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {4, 5, -2, 0, 0});
        assertEquals("f(x) = -2x^2 + 5x + 4", polynom.toString());
    }

    @Test
    void testHoles1() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {0, 1, -2, 0, 0});
        assertEquals("f(x) = -2x^2 + x", polynom.toString());
    }

    @Test
    void testHoles2() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {1, 0, -2, 0, 0});
        assertEquals("f(x) = -2x^2 + 1", polynom.toString());
    }

    @Test
    void testDegree() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {1, 0, 2, 0, 0});
        assertEquals(2, polynom.getDegree());
    }

    @Test
    void testPunktsymmetrie() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {0, 1, 0, 4, 0});
        assertTrue(polynom.isPointsymmetric());
    }

    @Test
    void testAchsensymmetrie() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {1, 0, 2, 0, 0});
        assertTrue(polynom.isAxissymmetric());
    }

    @Test
    void testKeinSymmetrie() throws WrongInputSizeException {
        Polynom polynom = new Polynom(new double[] {1, 4, 2, 0, 0});
        assertFalse(polynom.isPointsymmetric());
        assertFalse(polynom.isAxissymmetric());
    }

}

File: PolynomPython/src/TestPolynom.py
import unittest

from Polynom import Polynom


class MyTestCase(unittest.TestCase):
    def test_input_1(self):
        polynom = Polynom([5, 0, 0, 0, 0])
        self.assertEqual("f(x) = 5", polynom.__str__())

    def test_input_2(self):
        polynom = Polynom([5, 2, 0, 0, 0])
        self.assertEqual("f(x) = 2x + 5", polynom.__str__())

    def test_input_3(self):
        polynom = Polynom([5, 3, 4, 0, 0])
        self.assertEqual("f(x) = 4x^2 + 3x + 5", polynom.__str__())

    def test_input_4(self):
        polynom = Polynom([5, 1, 5, 6, 0])
        self.assertEqual("f(x) = 6x^3 + 5x^2 + 1x + 5", polynom.__str__())

    def test_input_5(self):
        polynom = Polynom([5, 2, 8, 9, 10])
        self.assertEqual("f(x) = 10x^4 + 9x^3 + 8x^2 + 2x + 5", polynom.__str__())

    def test_minus(self):
        polynom = Polynom([5, -2, 8, 9, 10])
        self.assertEqual("f(x) = 10x^4 + 9x^3 + 8x^2 - 2x + 5", polynom.__str__())

    def test_minus_at_the_start(self):
        polynom = Polynom([-5, 2, 8, 9, -10])
        self.assertEqual("f(x) = -10x^4 + 9x^3 + 8x^2 + 2x - 5", polynom.__str__())

    def test_holes_1(self):
        polynom = Polynom([0, 1, -2, 0, 0])
        self.assertEqual("f(x) = -2x^2 + 1x", polynom.__str__())

    def test_holes_2(self):
        polynom = Polynom([1, 0, -2, 0, 0])
        self.assertEqual("f(x) = -2x^2 + 1", polynom.__str__())

    def test_degree(self):
        polynom = Polynom([1, 0, 2, 0, 0])
        self.assertEqual(2, polynom.get_degree())

    def test_pointsymmetry(self):
        polynom = Polynom([0, 1, 0, 4, 0])
        self.assertEqual(polynom.SYMMETRIES[1], polynom.get_symmetry())

    def test_axissymmetry(self):
        polynom = Polynom([1, 0, 2, 0, 0])
        self.assertEqual(polynom.SYMMETRIES[0], polynom.get_symmetry())

    def test_no_symmetry(self):
        polynom = Polynom([1, 4, 2, 0, 0])
        self.assertEqual(polynom.SYMMETRIES[2], polynom.get_symmetry())


if __name__ == '__main__':
    unittest.main()
